src/cursor/cur_file.c

```diff
@@ -600,11 +593,47 @@ __curfile_reopen(WT_CURSOR *cursor, bool check_only)
         WT_ASSERT(session, dhandle->type == WT_DHANDLE_TYPE_BTREE);
         WT_ASSERT(session, ((WT_BTREE *)dhandle->handle)->root.page != NULL);
 
-        btree = CUR2BT(cbt);
+        btree = CUR2BT(cursor);
         cursor->internal_uri = btree->dhandle->name;
         cursor->key_format = btree->key_format;
         cursor->value_format = btree->value_format;
+
+        WT_STAT_CONN_DATA_INCR(session, cursor_reopen);
+    }
+    return (ret);
+}
+
+/*
+ * __curfile_reopen --
+ *     WT_CURSOR->reopen method for the btree cursor type.
+ */
+static int
+__curfile_reopen(WT_CURSOR *cursor, bool check_only)
+{
+    WT_DATA_HANDLE *dhandle;
+    WT_DECL_RET;
+    WT_SESSION_IMPL *session;
+    bool can_reopen;
+
+    session = CUR2S(cursor);
+    dhandle = ((WT_CURSOR_BTREE *)cursor)->dhandle;
+
+    if (check_only) {
+        can_reopen = WT_DHANDLE_CAN_REOPEN(dhandle);
+        /*
+         * If the data handle can't be reopened, it is a candidate for sweep, and that should never
+         * happen if any session (including ours) is actively using it.
+         */
+        WT_ASSERT(session, can_reopen || dhandle != session->dhandle);
+        return (can_reopen ? 0 : WT_NOTFOUND);
     }
+
+    /*
+     * Temporarily set the session's data handle. Reopen may be called either as part of an open API
+     * call, or during cursor sweep as part of a different API call, so we need to restore the data
+     * handle.
```

Comment:
Minor clarification on the comment. It seems like the first sentence and the last couple of words refer to different data handles? I'm not entirely sure. Maybe something like this (if it is correct):

Temporarily set the session's data handle to the dhande in this cursor. Reopen may be called... different API call, so we need to restore the original data handle that was in our session after reopen completes.

Reply:
Thanks, you're right.  I updated the comment to your wording.